department-of-veterans-affairs/caseflow
# MattT/APPEALS-10821: Alteration of VHA Benefit Type Selection for Non-VHA Users

Resolves [APPEALS-10821](https://vajira.max.gov/browse/APPEALS-10821)

[Diff Link](https://github.com/department-of-veterans-affairs/caseflow/compare/feature/APPEALS-8982..MattT/APPEALS-10821) | [Designs Link](https://bipuiux.invisionapp.com/console/share/QTCKRDXAHEN/953715692/play)

### Description
On the Review page for claim review intakes (HLRs and SCs) the "Veterans Health Administration" benefit type radio field option will be disabled if the user is not a member of the VHA business line. In addition, if the non-VHA user hovers over the VHA benefit type option they will be presented with an information tooltip.

### Acceptance Criteria
  - [x] A non-VHA user (user that is not a member of the Veterans Health Administration business line organization in Caseflow) will be able to view the following benefit types when intaking Higher-Level Reviews and Supplemental Claims: 
    - [x] Compensation 
    - [x] Pension & Survivor’s Benefits 
    - [x] Fiduciary  
    - [x] Insurance 
    - [x] Education 
    - [x] Veterans Readiness and Employment 
    - [x] Loan Guaranty  
    - [x] National Cemetery Administration  
    - [x] Veteran Health Administration
  - [x] Order of benefit types is changed to place Veterans Health Administration after National Cemetery Organization when user is on the Review page  
  - [ ] The following occurs when a non-VHA user attempts to select Veterans Health Administration as a benefit type: 
    - [x] Option will not be selectable 
    - [x] Option will be grayed out in its entirety  
    - [x] When hovered over, a message will appear with the following informational text:  
      - [x] Only VHA case team members can establish Higher Level Reviews and Supplemental Claims with VHA issues. If you have a VHA claim, please return the packet to VHA queue in the Centralized Mail Portal or send downloaded documents to VHABENEFITAPPEALS@va.gov.

### Testing Plan

1. Open the Rails console and add an Intake user to the VHA business line:
```ruby
BusinessLine.find_by(url: "vha").add_user(User.find_by_css_id("BVAISHAW"))
```

2. Log into Caseflow as this user. Navigate to Caseflow Intake ([Local link](http://localhost:3000/intake))

3. Start a Higher-Level Review intake.

4. Ensure in the "What is the Benefit Type?" section that the "Veterans Health Administration" radio field is enabled and selectable.

5. Cancel the intake, and repeat steps 3 and 4 for a Supplemental Claim intake.

6. Log into Caseflow as an Intake user that is **not** a member of the VHA business line (ex: `BVADWISE` should work).

7. Start another Higher-Level Review intake.

8. This time, the "Veterans Health Administration" option should be **disabled**. Ensure that it is not selectable.

9. Hover your cursor over the VHA benefit type option. A tooltip should appear. [Click here](https://bipuiux.invisionapp.com/console/share/QTCKRDXAHEN/953715692/play) for the associated design mockups to compare its appearance with what was specified.

10. Cancel this intake and repeat steps 7-9 for a Supplemental Claim.


### User Facing Changes

 USER|BEFORE|AFTER
 ---|---|---
 VHA|<img width="736" alt="before" src="https://user-images.githubusercontent.com/99351305/200997433-9eae045c-5d1a-409d-badb-dffe1f064c00.PNG">|<img width="738" alt="vha-after" src="https://user-images.githubusercontent.com/99351305/200997483-a7068362-2ba5-4a20-a27b-c384bd25e2b7.PNG">
 Non-VHA|<img width="736" alt="before" src="https://user-images.githubusercontent.com/99351305/200997433-9eae045c-5d1a-409d-badb-dffe1f064c00.PNG">|<img width="737" alt="vha-after" src="https://user-images.githubusercontent.com/99351305/201206926-7e69d3b2-9c59-4e2c-8716-ff8dce04144e.PNG">

### Code Documentation Updates

### Storybook Story
*For Frontend (Presentationa) Components*

<img width="650" alt="storybook" src="https://user-images.githubusercontent.com/99351305/201207461-86531f3e-22cd-42a5-9c68-fb48dfe39bde.PNG">
